src/network/networkd-link.c

```diff
@@ -2820,6 +2834,22 @@ int link_update(Link *link, sd_netlink_message *m) {
                         return r;
         }
 
+        /* find whether link is enslaved */
+        r = sd_netlink_message_read_u32(m, IFLA_MASTER, &master);
+        if (r >= 0) {
+
+                r = sd_netlink_message_enter_container(m, IFLA_PROTINFO);
+                if (r >= 0) {
+                        uint8_t state;
+
+                        r = sd_netlink_message_read_u8(m, IFLA_BRPORT_STATE, &state);
+                        if (r >= 0)
+                                link->bridge_port = true;
```

Comment:
I don't think we will reach here at first place.  In case of bond `IFLA_MASTER` will be true and there it wont find 
`IFLA_PROTINFO`. Bridge it's always there.

by default it's false. Not sure if we remove a interface we get any information. will look .